app/models/course_data/articles_courses.rb

```diff
@@ -131,12 +131,12 @@ def self.destroy_invalid_records(course, valid_article_ids)
     end
   end
 
-  def self.get_revisions_by_namespaces(course)
-    # Return revisions corresponding to tracked wikis and namespaces
+  def self.article_ids_by_namespaces(course)
+    # Return article ids from revisions corresponding to tracked wikis and namespaces
     course.tracked_namespaces.map do |wiki_ns|
       wiki = wiki_ns[:wiki]
       namespace = wiki_ns[:namespace]
       course.revisions.joins(:article).where(articles: { wiki: wiki, namespace: namespace })
-    end.flatten
+    end.flatten.pluck(:article_id)
```

Comment:
I think doing it this way will still load all the ActiveRecord objects into an array, I think (because of `.flatten`). I suggest building an array of ids more explicitly, by initializing an empty array and then adding ids to on each pass through the `tracked_namespaces` loop.